### PR TITLE
feat: Latest Activity Rail

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [ONYX-296] 

### Description

This PR adds the "Latest Activity" rail to the home screen (behind a feature flag).

Screenshot 2023-09-18 at 12 36 47

### Analytic Events

**Header:**

```
{
  "action": "tappedActivityGroup",
  "context_module": "activityRail",
  "context_screen_owner_type": "home",
  "destination_screen_owner_type": "activities",
  "type": "header"
}
```

**Item:**

```
{
  "action": "tappedActivityGroup",
  "context_module": "activityRail",
  "context_screen_owner_type": "home",
  "destination_screen_owner_type": "artist",
  "horizontal_slide_position": 0,
  "module_height": "single",
  "type": "thumbnail"
}
```

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Latest Activity Rail on home screen (behind feature flag) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Activity Screen refactoring - ole

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-296]: https://artsyproduct.atlassian.net/browse/ONYX-296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 08e85046efadf821f467c90dc77619fa11d88c64
</p>

File: src/app/Scenes/Activity/ActivityItem.tsx [hunk 1]
```diff
@@ -48,43 +30,12 @@ export const ActivityItem: React.FC<ActivityItemProps> = (props) => {
...
-        passProps: {
-          predefinedFilters: [sortFilterItem] as FilterArray,
-          searchCriteriaID: parsed.search_criteria_id,
-        },
-      })
-
     tracking.trackEvent(tracks.tappedNotification(item.notificationType))
 
-    navigateToActivityItem()
+    navigateToActivityItem(item.targetHref)
```
Comment (olerichter00) [inline src/app/Scenes/Activity/ActivityItem.tsx:35]: Extracting the logic to reuse it in the new rail.

File: src/app/Scenes/Activity/ActivityItem.tsx [hunk 2]
```diff
@@ -48,43 +30,12 @@ export const ActivityItem: React.FC<ActivityItemProps> = (props) => {
...
-          updater(item.id, store)
-        },
-        updater: (store) => {
-          updater(item.id, store)
-        },
-        onError: (error) => {
-          captureMessage(error?.stack!)
-        },
-      })
+      markAsRead(item)
```
Comment (olerichter00) [inline src/app/Scenes/Activity/ActivityItem.tsx:38]: Same here.

File: src/app/Scenes/Home/Components/ActivityRail.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  return (
+    <Flex pt={2}>
+      <Flex px={2}>
+        <SectionTitle fontWeight="bold" title={title} onPress={handleHeaderPress} />
+      </Flex>
+
+      <FlatList
+        horizontal
+        showsHorizontalScrollIndicator={false}
+        ListHeaderComponent={() => <Spacer x={2} />}
```
Comment (MounirDhahri) [inline src/app/Scenes/Home/Components/ActivityRail.tsx:60]: nice way to set in an initial horizontal padding 🙌

**Review by MounirDhahri — APPROVED** [review]
Nice PR!

**Review by dariakoko — DISMISSED** [review]
Looks good ⭐

**Pull request merged**